File: 251002/Nikonorova/lab1_rest/Repositories/CreatorRepository.cs
using distcomp.Models;
using System.Collections.Generic;
using System.Linq;

namespace distcomp.Repositories
{
    public class CreatorRepository : ICrudRepository<Creator, long>
    {
        private readonly Dictionary<long, Creator> _creators = new();
        private long _idCounter = 0;

        public Creator Save(Creator creator)
        {
            creator.Id = ++_idCounter;
            _creators[creator.Id] = creator; 
            return creator;
        }

        public Creator FindById(long id)
        {
            return _creators.TryGetValue(id, out var creator) ? creator : null;
        }

        public List<Creator> FindAll()
        {
            return _creators.Values.ToList();
        }

        public Creator Update(Creator creator)
        {
            if (_creators.ContainsKey(creator.Id))
            {
                _creators[creator.Id] = creator; 
                return creator;
            }
            return null;
        }

        public bool DeleteById(long id)
        {
            return _creators.Remove(id); 
        }

    }
}

File: 251004/Kalitsko/1lab/Rest/src/main/java/com/example/rest/repository/CreatorRepository.java
package com.example.rest.repository;

import com.example.rest.entity.Creator;

import java.util.List;
import java.util.Optional;

public interface CreatorRepository {

    Creator create(Creator creator);
    Creator update(Creator updatedCreator);
    void deleteById(Long id);
    List<Creator> findAll();
    Optional<Creator> findById(Long id);
}
